tests/src/test/scala/org/http4s/parser/AuthorizationHeaderSpec.scala

```diff
@@ -1,26 +1,34 @@
-package org.http4s.parser
+package org.http4s
+package parser
 
-import org.http4s.{Http4sSpec, OAuth2BearerToken}
 import org.http4s.headers.Authorization
-
+import org.http4s.util.NonEmptyList
+import org.http4s.util.string._
 
 class AuthorizationHeaderSpec extends Http4sSpec {
 
   def hparse(value: String) = HttpHeaderParser.AUTHORIZATION(value)
 
   "Authorization header" should {
-    "Parse a valid Oauth2 header" in {
+    "Parse a valid OAuth2 header" in {
       val token = (('a' to 'z') ++ ('A' to 'Z') ++ ('0' to '9') ++ "-._~+/".toSeq).mkString
-      val h = Authorization(OAuth2BearerToken(token + "="))
+      val h = Authorization(Credentials.Token(AuthScheme.Bearer, token + "="))
       hparse(h.value) must be_\/-(h)
     }
 
-    "Reject an ivalid Oauth2 header" in {
+    "Reject an invalid OAuth2 header" in {
       val invalidTokens = Seq("f!@", "=abc", "abc d")
       forall(invalidTokens) { token =>
-        val h = Authorization(OAuth2BearerToken(token))
+        val h = Authorization(Credentials.Token(AuthScheme.Bearer, token))
```

Comment:
The fact that this is allowed is kind of broken: any Credentials that we create should round trip through render and parse.  Creating credentials from a String, instead of a verified token, is what gives rise to this.  The question is how to balance safety and usability.  The two obvious choices are a `token` type or a `parse: ParseResult` constructor.